packages/dotnet/ReadMe/HarJsonTranslationLogics/RequestProcessor.cs

```diff
@@ -26,8 +26,16 @@ public async Task<Request> ProcessRequest()
       requestObj.cookies = this.GetCookies();
       requestObj.method = this.request.Method;
       requestObj.url = this.request.Scheme + "://" + this.request.Host.Host + ":" + this.request.Host.Port + this.request.Path;
+      requestObj.url += this.request.QueryString.ToString();
+
       requestObj.httpVersion = this.request.Protocol;
-      requestObj.postData = await this.GetPostData();
+
+      // We should ony add POST data into the HAR if we have it.
+      PostData postData = await this.GetPostData();
```

Comment:
Without this change we were setting `postData` to `{}` on GET requests.  However if I only added the line below this to conditionally add `postData` if we had it then `postData` would get sent to us as `null` because though the `PostData` class tells the JSON serializer to ignore null values, the `Request` class didn't.